devito/builtins/arithmetic.py

```diff
@@ -58,6 +57,57 @@ def norm(f, order=2):
     return f.dtype(v)
 
 
+@dv.switchconfig(log_level='ERROR')
+def sum(f, dims=None):
+    """
+    Compute the sum of the Function data over specified dimensions.
+    Defaults to sum over all dmensions
+
+    Parameters
+    ----------
+    f : Function
+        Input Function.
+    dims : Dimension or tuple of Dimension
+        Dimensions to sum over.
+    """
+    dims = dv.tools.as_tuple(dims)
+    if dims == () or dims == f.dimensions:
+        return sumall(f)
+
+    # Get dimensions and shape of the result
+    new_dims = tuple(d for d in f.dimensions if d not in dims)
+    shape = tuple(f._size_domain[d] for d in new_dims)
+    if f.is_TimeFunction and f.time_dim not in dims:
+        out = f._rebuild(name="%ssum" % f.name, shape=shape, dimensions=new_dims,
+                         initializer=np.empty(0))
+    elif f.is_SparseTimeFunction:
+        if f.time_dim in dims:
+            # Sum over time -> SparseFunction
+            coords = f.coordinates._rebuild(name="%ssum_coords" % f.name)
```

Comment:
I would rename this as new_coords to be consistent with new_dims above...or rename new_dims to dims